chat-server/src/services/embeddings.ts

```diff
@@ -1,42 +1,95 @@
-import { Configuration, OpenAIApi } from "openai";
+import { OpenAIClient } from "../integrations/openai";
+import { logger } from "./logger";
+import dotenv from "dotenv";
+import { stripIndent } from "common-tags";
+dotenv.config();
+
+interface EmbeddingParams {
+  text: string;
+  userIp: string;
+}
+
+interface EmbeddingResponse {
+  status: number;
+  errorData?: any;
+  embeddings?: number[];
+}
 
 class EmbeddingService {
   private embeddingProvider: EmbeddingProvider;
   constructor(embeddingProvider: EmbeddingProvider) {
     this.embeddingProvider = embeddingProvider;
   }
 
-  async createEmbedding(text: string) {
-    return await this.embeddingProvider.createEmbedding(text);
+  async createEmbedding({ text, userIp }: EmbeddingParams) {
+    return await this.embeddingProvider.createEmbedding({ text, userIp });
   }
 }
 
 // Abstract interface for embedding provider to make it easier to swap out
 // different providers in the future.
 abstract class EmbeddingProvider {
```

Comment:
Why abstract class? Why not just exported interface?

Reply:
b/c i've been writing a lot of Dart and Kotlin recently 😄 

will refactor

Reply:
also updated `LlmProvider` in `llm.ts` to follow same pattern